2.4/contrib/bin/run.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+# SCL in CentOS/RHEL 7 doesn't support --exec, we need to do it ourselves
+source scl_source enable mongodb24
+set -e
+
+function usage {
+	echo "You must specify following environment variables:"
+	echo "  \$MONGODB_USER"
+	echo "  \$MONGODB_PASSWORD"
+	echo "  \$MONGODB_DB - optional (default database - 'production')"
+	echo "  \$MONGODB_ADMIN_PASSWORD - optional"
+	exit 1
+}
+
+function create_mongodb_users {
+	mongod -f /opt/openshift/etc/mongodb.conf &
+
+	# Check if the MongoDB daemon is up.
+	RC=1
+	while [[ RC -ne 0 ]]; do
+	    echo "=> Waiting for confirmation of MongoDB service startup"
+	    sleep 2
```

Comment:
also same comment as below, if mongodb gets stuck and will not accept connections for some reason (even though it will show itself as running, this will be waiting forever to start (and the container might still be reporting as 'running').